src/lib/blueprint.ts

```diff
@@ -357,13 +366,80 @@ const createRoute = async (
     throw new Error(`Could not resolve name for route at ${path}`)
   }
 
+  const endpoints = await createEndpoints(path, pathItem, context)
+
   return {
     path: routePath,
     name,
-    namespace: namespace != null ? { path: namespace } : null,
-    endpoints: await createEndpoints(path, pathItem, context),
+    namespace:
+      namespace != null
+        ? {
+            path: namespace,
+            isDeprecated: false,
+            isUndocumented: false,
+          }
+        : null,
+    endpoints,
     subroutes: [],
+    isUndocumented: false,
+    isDeprecated: false,
+  }
+}
+
+const updateRoutesStatus = (routes: Route[]): Route[] => {
+  return routes.map((route) => {
+    const isRouteDeprecated = route.endpoints.every(
+      (endpoint) => endpoint.isDeprecated,
+    )
+    const isRouteUndocumented = route.endpoints.every(
+      (endpoint) => endpoint.isUndocumented,
+    )
+
+    return {
+      ...route,
+      isDeprecated: isRouteDeprecated,
+      isUndocumented: isRouteUndocumented,
+    }
+  })
+}
+
+const updateNamespaceStatus = (routes: Route[]): Route[] => {
+  // Group routes by namespace
+  const namespaceGroups = routes.reduce<Record<string, Route[]>>(
```

Comment:
Can we factor this our into a function, `getRoutesByNamespace` (this removes the need for the comment)?